### PR TITLE
LPS-133722 Sort intances by asset type

### PR DESCRIPTION
**Pull request by MarinhoFeliphe**

### CODE REVIEW

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/odata/entity/v1_0/InstanceEntityModel.java [hunk 1]
```diff
@@ -30,6 +31,7 @@
 
 	public InstanceEntityModel() {
 		_entityFieldsMap = Stream.of(
+			new StringEntityField("assetType", locale -> "assetType"),
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/odata/entity/v1_0/InstanceEntityModel.java:34]: Hey @MarinhoFeliphe, as it is a internationalized field use `new StringEntityField(
				"assetType", locale -> Field.getSortableFieldName("assetType"))` instead. Thx

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java [hunk 2]
```diff
@@ -1131,7 +1133,16 @@ private FieldSort _toFieldSort(Sort[] sorts) {
 
 		Sort sort = (Sort)ArrayUtil.getValue(sorts, 0);
 
-		if (StringUtil.equals(sort.getFieldName(), "createDate")) {
+		if (StringUtil.equals(sort.getFieldName(), "assetType")) {
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/InstanceResourceImpl.java:1136]: Hey @MarinhoFeliphe, adapt these changes here to be compatible with the proposed change above(InstanceEntityModel.java:34). Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]
Hey @MarinhoFeliphe, just to check. Do the generated tests cover this new sort option? thx

**Comment by rafaprax** [comment]
ci:stop

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/odata/entity/v1_0/InstanceEntityModel.java (hunk 1 above)
Comment (rafaprax) [reply modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/odata/entity/v1_0/InstanceEntityModel.java:34]: just fixing an error in the proposed solution, use `new StringEntityField(
				"assetType",
				locale -> Field.getSortableFieldName(
					"localized_assetType_" + LocaleUtil.toLanguageId(locale))));`

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:stop

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Review by rafaprax — APPROVED** [review]

**Comment by rafaprax** [comment]
ci:test:relevant

**Comment by rafaprax** [comment]
ci:stop

**Comment by ramonmsp** [comment]
Just started reviewing :)

**Comment by rafaprax** [comment]
These changes will be sent as part of https://github.com/liferay-workflow/liferay-portal/pull/395

**Pull request closed without merging**